### DISCUSSION TITLE
Tutorial step 18 - resetting a shot profile

### DISCUSSION THREAD

**Discussion by ttrickel** [missionpinball/missionpinball#71] [Q&A]
I've reached the end of Tutorial step 18: Add your first shot.

The tutorial ends with the creation of a shot profile and that works great - as far as it goes.

What I'd like to do is reset the shot profile on mode start or on mode stop. Unfortunately, I can't find the correct syntax.

It seems something like the following should work but it doesn't. What am I missing?

shots:
  my_first_shot_mode2:
    switch: s_reentry_left
    show_tokens:
      led: l_re-entry_left
    profile: mode2

shot_profiles:
  mode2:
    states:
      - name: flashing
        show: flash
        speed: 5
      - name: lit
        show: on
    loop: false
    block: true

event_player:
  mode_mode2_stopped:
    my_first_shot_mode2_state_name:
        flashing

**Comment by atummons** [comment]
https://docs.missionpinball.org/en/stable/config/shots.html#restart-events

Not saying this is the best way for advanced logic. But you can add this to your shots: section for that shot. Then add the mode_started (put your modes name). Then every time the mode starts, it will reset the profile to its first state and enable the shot.

**Reply by ttrickel** [reply]
Thanks for the quick response. But, I must be really dense.

Is this what you meant? If so, it doesn't work for me. The state_name stays at lit in the events monitor and the led just stays on solid.

shots:
  my_first_shot_mode2:
    switch: s_reentry_left
    show_tokens:
      led: l_re-entry_left
    profile: mode2
    restart_events: mode2_started
